### PR TITLE
chore: upgrade to tenderdash v0.13

### PR DESCRIPTION
**Pull request by shotonoff**
## Issue being fixed or feature implemented

Tenderdash v0.13-dev has incompatible changes that need to be supported in the ABCI platform

## What was done?

* update `rs-tenderdash-abci` to the latest version
* fix failed unit tests

## How Has This Been Tested?

Unit tests

## Breaking Changes

ABCI

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by shumkov — APPROVED** [review]
Looks good to me

File: Cargo.lock [hunk 1]
```diff
@@ -2115,7 +2115,7 @@ source = "git+https://github.com/fominok/jsonschema-rs?branch=feat-unevaluated-p
 dependencies = [
  "ahash 0.7.6",
  "anyhow",
- "base64 0.21.2",
+ "base64 0.13.1",
```
Comment (shumkov) [inline Cargo.lock:2118]: Why did you downgrade it?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: Cargo.lock (hunk 1 above)
Comment (shotonoff) [reply Cargo.lock:2118]: it is automatically, once I've run `cargo update`

**Review by shumkov — APPROVED** [review]

**Comment by shotonoff** [comment]
2 failed tasks are not blocker to merging. locally, all tests are passed

**Pull request merged**